.github/workflows/pytest.yml

```diff
@@ -0,0 +1,68 @@
+name: Pytest
+on: [push, pull_request, workflow_dispatch]
+
+env:
+  HUGEPAGES: 512
+
+jobs:
+  pytest:
+    strategy:
+      matrix:
+        test: ["cli", "state", "multi_gateway"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          # git submodule update --init --recursive
+          submodules: recursive
+
+      - name: Build images
+        run: |
+          SVC="ceph nvmeof-devel" make build
+
+      - name: Setup huge pages
+        run: |
+          make setup HUGEPAGES=$HUGEPAGES
+
+      - name: Start ceph cluster
+        run: |
+          SVC=ceph OPTS="--detach" make up
+
+      - name: Wait for the ceph cluster container to become healthy
+        run: |
+          while true; do
+            container_status=$(docker inspect --format='{{.State.Health.Status}}' ceph)
+            if [[ $container_status == "healthy" ]]; then
+            break
+          else
+            # Wait for a specific time before checking again
+            sleep 1
+            echo -n .
+          fi
+          done
+          echo
+
+      - name: Create RBD image
+        run: |
+          echo "💁 ceph list pools:"
+          SVC=ceph OPTS="-T" CMD="ceph osd lspools" make exec
+          echo "💁 rbd create:"
+          SVC=ceph OPTS="-T" CMD="rbd create rbd/mytestdevimage --size 16" make exec
+          echo "💁 ls rbd:"
+          SVC=ceph OPTS="-T" CMD="rbd ls rbd" make exec
+
+      - name: Run ${{ matrix.test }} test
+        run: |
+          # Run tests code in current dir
+          SVC="nvmeof-devel" OPTS="--volume=$(pwd)/tests:/src/tests --entrypoint=python3" CMD="-m pytest -s tests/test_${{ matrix.test }}.py" make run
```

Comment:
@epuertat, Trying a single workflow with a common build images job, using GitHub artifacts to share the container images between the jobs. but I can't get quite to _"one matrix to rule them all"_ since the demo environment is slightly different compared to pytest one, most notably running vs. embedding of the gateway. WDYT?



```mermaid
graph LR
    B(build) -->D[demo]
    B -->P[pytest matrix]
    P-->C(cli)
    P-->S(state)
    P-->M(multi gateway)
```

Reply:
OM🐶, this is awesome @baum!

![image](https://github.com/ceph/ceph-nvmeof/assets/37327689/5f5a6e72-1854-4679-89f1-148e55cb0d1d)

I was checking the Github Packages quotas (container registry) and unfortunately those are pretty limited (1000 GB monthly data transfer), but... for Github Actions that's unlimited, so we can push images there.

BTW, I was playing with the Github Actions cache, and it can speed things up a lot (it's like the local container cache). The main issue is that the cache is not shared across PRs, so speed-ups are only noticed from the 2nd CI run onwards.